### PR TITLE
[BEAM-551] Add better toString methods for VPs

### PR DESCRIPTION
**Pull request by sammcveety**
R: @davorbonaci

### CODE REVIEW

File: sdks/java/core/src/main/java/org/apache/beam/sdk/options/ValueProvider.java [hunk 1]
```diff
@@ -91,6 +91,11 @@ public T get() {
     public boolean isAccessible() {
       return true;
     }
+
+    @Override
+    public String toString() {
+      return String.format("StaticValueProvider: %s", value);
```
Comment (dhalperi) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/options/ValueProvider.java]: Use [`MoreObjects.toStringHelper`](https://github.com/google/guava/wiki/CommonObjectUtilitiesExplained#tostring)?

Unified format, no need to make decisions on sentences, etc. And IDE renames kind of auto-update ToStrings.

```java
return MoreObjects.toStringHelper(StaticValueProvider.class)
    .add("value", value)
    .toString();
```

Comment (dhalperi) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/options/ValueProvider.java]: I think that comes out roughly as "StaticValueProvider{value=value)".

File: sdks/java/core/src/main/java/org/apache/beam/sdk/options/ValueProvider.java [hunk 2]
```diff
@@ -212,6 +222,12 @@ public boolean isAccessible() {
     public String propertyName() {
       return propertyName;
     }
+
+    @Override
+    public String toString() {
+      return String.format(
+          "RuntimeValueProvider from property %s", propertyName);
```
Comment (dhalperi) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/options/ValueProvider.java]: if accessible, also show the value?

**Comment by sammcveety** [comment]
Done, please take another look.

**Comment by davorbonaci** [comment]
LGTM.

**Comment by davorbonaci** [comment]
Failures are unrelated. Merging.

**Pull request closed without merging**